### DISCUSSION TITLE
Omitted rows from filtering replaced with NAs when viewing a dataframe

### DISCUSSION THREAD

**Discussion by rmcchrystal** [posit-dev/positron#4045] [Feedback] [answered]
Just a minor observation. I have noticed that when I have a dataframe open in the viewer and apply filtering to it, the remaining rows from before filtering are replaced with NAs, rather than being removed completely. This is easily resolved by closing and re-opening the dataframe in the viewer.

Not something that takes away from using Positron, but potentially a minor fix for future releases?

**Comment by jmcphers** [comment]
This is definitely unexpected, and in my testing I couldn't reproduce it. Could you share an example?

**Comment by dfalbel** [comment]
Maybe related to https://github.com/posit-dev/positron/issues/4170 ?

**Comment by dfalbel** [comment, marked as answer]
I was able to reproduce this with:

```
x <- data.frame(x = c(1,2,3))
View(x)
# now add a filter for eg x > 1
x$x <- NA_real_
```

Going back to viewer, there are still 2 lines, and they show `NA`.

I can also confirm that the fix for #4170 will also fix this issue.

**Reply by rmcchrystal** [reply]
Apologies for not replying to this, been away on holidays. Thanks @dfalbel for this! Marking as answered.
